java/client/src/main/java/glide/api/RedisClusterClient.java

```diff
@@ -0,0 +1,67 @@
+package glide.api;
+
+import static glide.api.RedisClient.buildChannelHandler;
+import static glide.api.RedisClient.buildCommandManager;
+import static glide.api.RedisClient.buildConnectionManager;
+
+import glide.api.commands.ClusterBaseCommands;
+import glide.api.models.ClusterValue;
+import glide.api.models.configuration.RedisClusterClientConfiguration;
+import glide.api.models.configuration.Route;
+import glide.connectors.handlers.ChannelHandler;
+import glide.managers.CommandManager;
+import glide.managers.ConnectionManager;
+import glide.managers.models.Command;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Async (non-blocking) client for Redis in Cluster mode. Use {@link #CreateClient} to request a
+ * client to Redis.
+ */
+public class RedisClusterClient extends BaseClient implements ClusterBaseCommands {
+
+    protected RedisClusterClient(ConnectionManager connectionManager, CommandManager commandManager) {
+        super(connectionManager, commandManager);
+    }
+
+    /**
+     * Async request for an async (non-blocking) Redis client in Cluster mode.
+     *
+     * @param config Redis cluster client Configuration
+     * @return a Future to connect and return a ClusterClient
+     */
+    public static CompletableFuture<RedisClusterClient> CreateClient(
+            RedisClusterClientConfiguration config) {
+        try {
+            ChannelHandler channelHandler = buildChannelHandler();
```

Comment:
maybe the parts repeated here from RedisClient should be offloaded to BaseClient?

Reply:
Updated in 1ea00600

Reply:
Reworked in 13458bf6